From pearl-miui-camera-compat
Subject: [PATCH] Fix rebuilt resource and beauty fallbacks

The rebuilt package needs to retain the encrypted-resource loader contract.
Also tolerate a beauty type not present in this device configuration instead of
terminating the app.

---
 smali/com/android/camera/fragment/FragmentBeauty.smali | 15 +++------------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/smali/com/android/camera/fragment/FragmentBeauty.smali b/smali/com/android/camera/fragment/FragmentBeauty.smali
index pearl..compat 100644
--- a/smali/com/android/camera/fragment/FragmentBeauty.smali
+++ b/smali/com/android/camera/fragment/FragmentBeauty.smali
@@ -2016,17 +2016,9 @@
     goto/16 :goto_2

     :cond_1e
-    new-instance p0, Ljava/lang/RuntimeException;
-
-    const-string/jumbo p1, "unknown beauty type: "
-
-    invoke-virtual {p1, v7}, Ljava/lang/String;->concat(Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p1
-
-    invoke-direct {p0, p1}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
-
-    throw p0
+    invoke-virtual {v4, v8}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
+
+    goto/16 :goto_2

     :cond_1f
     invoke-virtual {p0, v4}, Lcom/android/camera/fragment/FragmentViewPagerContainer;->setPagerFragmentContainer(Ljava/util/List;)V
